core/templates/dev/head/domain/statistics/LearnerActionObjectFactory.js

```diff
@@ -17,63 +17,71 @@
  *     Action domain objects.
  */
 
-oppia.factory('LearnerActionObjectFactory', [function() {
-  /**
-   * @constructor
-   * @param {string} actionType - type of an action.
-   * @param {Object.<string, *>} actionCustomizationArgs - customization dict
-   *   for an action.
-   * @param {number} schemaVersion - schema version of the class instance.
-   */
-  var LearnerAction = function(
-      actionType, actionCustomizationArgs, schemaVersion) {
-    /** @type {string} */
-    this.actionType = actionType;
-    /** @type {Object.<string, *>} */
-    this.actionCustomizationArgs = actionCustomizationArgs;
-    /** @type {number} */
-    this.schemaVersion = schemaVersion;
-  };
+oppia.constant('LEARNER_ACTION_SCHEMA_LATEST_VERSION', 1);
 
-  /**
-   * @property {string} actionType - type of an action
-   * @property {Object.<string, *>} actionCustomizationArgs - customization dict
-   *   for an action
-   * @property {number} schemaVersion - schema version of the class instance
-   * @returns {LearnerAction}
-   */
-  LearnerAction.createNew = function(
-      actionType, actionCustomizationArgs, schemaVersion) {
-    return new LearnerAction(
-      actionType, actionCustomizationArgs, schemaVersion);
-  };
+oppia.factory('LearnerActionObjectFactory', [
+  'LEARNER_ACTION_SCHEMA_LATEST_VERSION',
+  function(LEARNER_ACTION_SCHEMA_LATEST_VERSION) {
+    /**
+     * @constructor
+     * @param {string} actionType - type of an action.
+     * @param {Object.<string, *>} actionCustomizationArgs - customization dict
+     *   for an action.
+     * @param {number} schemaVersion - schema version of the class instance.
+     */
+    var LearnerAction = function(
+        actionType, actionCustomizationArgs, schemaVersion) {
+      /** @type {string} */
+      this.actionType = actionType;
+      /** @type {Object.<string, *>} */
+      this.actionCustomizationArgs = actionCustomizationArgs;
+      /** @type {number} */
+      this.schemaVersion = schemaVersion;
+    };
+
+    /**
+     * @property {string} actionType - type of an action
+     * @property {Object.<string, *>} actionCustomizationArgs - customization
+     *   dict for an action
+     * @property {number} [schemaVersion=LEARNER_ACTION_SCHEMA_LATEST_VERSION]
+     *   - schema version of the class instance.
+     * @returns {LearnerAction}
+     */
+    LearnerAction.createNew = function(
+        actionType, actionCustomizationArgs, schemaVersion) {
+      schemaVersion = schemaVersion || LEARNER_ACTION_SCHEMA_LATEST_VERSION;
```

Comment:
Do we need to check that schemaVersion is <= LEARNER_ACTION_SCHEMA_LATEST_VERSION?

Reply:
Hmmm, ok. But won't creating something with a schema version larger than the "latest" version be illegal? Ok to defer this to later if you feel so, but I think even if done later, the check would be what I mentioned above?